napari/components/layerlist.py

```diff
@@ -87,6 +101,8 @@ def insert(self, index: int, value: Layer):
         """Insert ``value`` before index."""
         new_layer = self._type_check(value)
         new_layer.name = self._coerce_name(new_layer.name)
+        self._clean_cache()
+        new_layer.events.set_data.connect(self._clean_cache)
```

Comment:
We've generally tried to avoid these types of event connects. I see you've added `_process_delete_item` to remove it, but very curious what @tlambert03 and @jni think about it. This all fells a little bug prone, but maybe it will be ok. I'd like to get some more opinions before merge though

Reply:
As I think about this the classical approach for this is to introduce parenthood relation and inside `Base._update_dims` call 

```python
if self._parent is not None:
	self._parent._clean_cache
```

But I'm not sure if this is not too big change?